src/fv/share/tdymeshcustom.c

```diff
@@ -0,0 +1,1315 @@
+
+        // Find the iface that is shared by icell and cell_id2
+        for (iface=0; iface<nfaces; iface++) {
+          face_id = cell_to_face[iface][icell];
+          for (PetscInt iside=0; iside<2; iside++) {
+            cell_id2 = face_to_cell[iside][face_id];
+            if (cell_id2 == abs(dual_id)) {
+              iface_tmp = iface;
+              found = PETSC_TRUE;
+              break;
+            }
+          }
+          if (found) break;
+        }
+
+        if (found) {
+          ugrid->connection_to_face[iconn] = face_id;
+        } else {
+          SETERRQ(PETSC_COMM_WORLD,PETSC_ERR_USER,"Face not found");
+        }
+        iface = iface_tmp;
+
+        // Check that there exist a correspond face in the cell_id2 for
+        // the iface-th of icell-th
+        found = PETSC_FALSE;
+        for (iface2=0; iface2<ugrid->cell_num_vertices[cell_id2]; iface2++) {
+          if (cell_to_face[iface][icell] == cell_to_face[iface2][cell_id2]) {
+            iface2_tmp = iface2;
+            found = PETSC_TRUE;
+            break;
+          }
+        }
+        iface2 = iface2_tmp;
+
+        // Now check the corresponding faces are of the same type
+        if (found) {
+          PetscInt num_cell_vertices = ugrid->cell_num_vertices[icell];
+          TDyCellType icell_type = GetCellType(num_cell_vertices);
+          TDyFaceType face_type = TDyGetFaceTypeForCellType(icell_type, iface);
+
+          PetscInt num_cell_vertices2 = ugrid->cell_num_vertices[cell_id2];
+          TDyCellType icell_type2 = GetCellType(num_cell_vertices2);
+          TDyFaceType face_type2 = TDyGetFaceTypeForCellType(icell_type2,iface2);
+
+          if (face_type != face_type2) {
+            SETERRQ(PETSC_COMM_WORLD,PETSC_ERR_USER,"Face types do not match");
+          }
+
+          PetscInt nvertices = TDyGetNumVerticesForFaceType(face_type);
+          mesh_ptr->faces.num_vertices[iconn] = nvertices;
+
+        } else {
+          SETERRQ(PETSC_COMM_WORLD,PETSC_ERR_USER,"Global face not found");
+        }
+
+        mesh_ptr->faces.cell_offset[iconn] = offset;
+        mesh_ptr->faces.cell_ids[offset++] = icell;
+        mesh_ptr->faces.cell_ids[offset++] = abs(dual_id);
+        mesh_ptr->faces.id[iconn] = cell_to_face[iface][icell];
+        mesh_ptr->faces.is_internal[iconn] = PETSC_TRUE;
+        mesh_ptr->faces.is_local[iconn] = PETSC_TRUE;
+        iconn++;
+      }
+    }
+  }
+
+  PetscFunctionReturn(0);
+}
+
+/// Computes area of triangle
+///
+/// @param [in] ugrid A TDyUGrid struct
+/// @param [in] v_id1 First vertex ID of the triangel
+/// @param [in] v_id2 Second vertex ID of the triangel
+/// @param [in] v_id3 Third vertex ID of the triangel
+/// @param [out] area Area of the triange
+///
+/// @returns 0 on success, or a non-zero error code on failure
+static PetscErrorCode ComputeArea(TDyUGrid *ugrid, PetscInt v_id1, PetscInt v_id2, PetscInt v_id3, PetscReal *area) {
```

Comment:
Consider renaming this to `ComputeTriArea` or something that indicates that we're computing the area of a triangle.

Reply:
Fixed as suggested.